File: 1334-find-the-city-with-the-smallest-number-of-neighbors-at-a-threshold-distance/1334-find-the-city-with-the-smallest-number-of-neighbors-at-a-threshold-distance.cpp
class Solution {
public:
    int findTheCity(int N, vector<vector<int>>& edges, int dt) {
        int ans = -1;
        vector<vector<int>> g(N, vector<int>(N, INT_MAX));

        for (int i = 0; i < N; i++) {
            g[i][i] = 0;
        }

        for (auto edge : edges) {
            int u = edge[0], v = edge[1], w = edge[2];
            g[u][v] = w;
            g[v][u] = w;
        }

        for (int k = 0; k < N; k++) {
            for (int i = 0; i < N; i++) {
                for (int j = 0; j < N; j++) {
                    if (g[i][k] != INT_MAX && g[k][j] != INT_MAX) {
                        g[i][j] = min(g[i][j], g[i][k] + g[k][j]);
                    }
                }
            }
        }

        int curr = INT_MAX;

        for (int i = 0; i < N; i++) {
            int temp = 0;
            for (int j = 0; j < N; j++) {
                if (g[i][j] <= dt) {
                    temp++;
                }
            }
            if (temp <= curr) {
                ans = i;
                curr = temp;
            }
        }

        return ans;
    }
};

File: 1334-find-the-city-with-the-smallest-number-of-neighbors-at-a-threshold-distance/1334-find-the-city-with-the-smallest-number-of-neighbors-at-a-threshold-distance.py
class Solution:
    def findTheCity(self, N: int, edges: List[List[int]], dt: int) -> int:
        ans = None
        g = [[float('inf')] * N for _ in range(N)]

        for i in range(N):
            g[i][i] = 0
            
        for u, v, w in edges:
            g[u][v] = w
            g[v][u] = w
            
        for k in range(N):
            for i in range(N):
                for j in range(N):
                    g[i][j] = min(g[i][j], g[i][k] + g[k][j])
                    
        curr = float('inf')

        for i in range(N):
            temp = 0
            for j in range(N):
                if g[i][j] <= dt:
                    temp += 1
            if temp <= curr:
                ans = i
                curr = temp
                
        return ans